### PR TITLE
Add tests for ArrayBuffer.prototype.transferToFixedLength

### PR DESCRIPTION
**Pull request by syg**
These are copied from ArrayBuffer/prototype/transfer/ with expectations adjusted.

See https://tc39.es/proposal-arraybuffer-transfer/

### CODE REVIEW

File: test/built-ins/ArrayBuffer/prototype/transferToFixedLength/new-length-excessive.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  1. Let O be the this value.
+  2. Perform ? RequireInternalSlot(O, [[ArrayBufferData]]).
+  3. If IsSharedArrayBuffer(O) is true, throw a TypeError exception.
+  4. If IsDetachedBuffer(O) is true, throw a TypeError exception.
+  5. If newLength is undefined, let newByteLength be
+     O.[[ArrayBufferByteLength]].
+  6. Else, let newByteLength be ? ToIntegerOrInfinity(newLength).
+  7. Let new be ? Construct(%ArrayBuffer%, « 𝔽(newByteLength) »).
+  [...]
```
Comment (ljharb) [inline test/built-ins/ArrayBuffer/prototype/transferToFixedLength/new-length-excessive.js:19]: these aren't the spec steps for transferToFixedLength: https://tc39.es/proposal-arraybuffer-transfer/ only has two steps, and they call into https://tc39.es/proposal-arraybuffer-transfer/, and the byte length is ToIndex'd which clamps it - so it shouldn't be throwing, i think?

Comment (ljharb) [reply test/built-ins/ArrayBuffer/prototype/transferToFixedLength/new-length-excessive.js:19]: (that suggests that the transfer tests also need updating)

Comment (ljharb) [reply test/built-ins/ArrayBuffer/prototype/transferToFixedLength/new-length-excessive.js:19]: aha, thanks for clarifying, you're right

**Review by ljharb — APPROVED** [review]
confirmed, everything looks good :-D

File: test/built-ins/ArrayBuffer/prototype/transferToFixedLength/new-length-non-number.js
Comment (ptomato) [inline test/built-ins/ArrayBuffer/prototype/transferToFixedLength/new-length-non-number.js:1]: This test looks like a good case that moving [`toPrimitiveObserver()`](https://github.com/tc39/test262/blob/main/harness/temporalHelpers.js#L1819-L1842) out of TemporalHelpers and into a general helper file would be useful for other proposals. Either way, that's out of scope for this PR, just noting it down for myself for future reference.

**Review by ptomato — APPROVED** [review]

File: test/built-ins/ArrayBuffer/prototype/transferToFixedLength/new-length-excessive.js (hunk 1 above)
Comment (syg) [reply test/built-ins/ArrayBuffer/prototype/transferToFixedLength/new-length-excessive.js]: I've removed the spec steps from both of those tests and changed the description to talk about clamping in ToIndex.

**Comment by syg** [comment]
Can this be merged?

**Pull request merged**